### DISCUSSION TITLE
1.21.0 breaks half functionality for XHK1-TC Xfinity Alarm security keypad

### DISCUSSION THREAD

**Discussion by ikke-zelf** [Koenkk/zigbee2mqtt#8629] [Devices]
Hi,
Since a little time ( I expect this to be as of release 1.21.0 as this is where I notice an entry in the release notes for the XHK1-TC Xfinity Alarm security keypad) the functionality for the keypad icw zigbee2mqtt has been stripped and changed.

Prior to this we had the following modes available:
disarm, arming_night, arming_away, arming_stay, entry_delay, exit_delay and in_alarm.
When one of the 'arming' modes was set the keypad would lit up the corresponding button light in red.
When the entry or exit_delay was set, the keypad would starts beeping until a different mode is set.

Now the supported modes are: disarm, arm_day_zones, arm_night_zones, arm_all_zones, invalid_code, not_ready, already_disarmed.
The buttons (top three) do not lit anymore, and the beeping is gone when setting (trying to) the exit_delay.

I use this functionality with node-red and home assistant and am not into this coding enough to put my finger on what to change in git to restore the functionality.

any help would be appreciated

**Comment by sjorge** [comment]
For those finding this post but not the related issue: https://github.com/Koenkk/zigbee2mqtt/issues/8630

The changes are clarified in the issue.
